src/cargo/core/package_id_spec.rs

```diff
@@ -139,8 +139,8 @@ impl PackageIdSpec {
                                        your project, and the specification \
                                        `{}` is ambiguous.\n\
                                        Please re-run this command \
-                                       with `-p <spec>` where `<spec>` is one \
-                                       of the following:",
+                                       with is one of the following as \
```

Comment:
Oh, I see now. I didn't connect that the `-p` doesn't work. No idea how I missed that in the description, sorry.